tests/unittest/test_batchify.py

```diff
@@ -5,18 +5,21 @@
 
 import pytest
 
+
 def test_pad():
     padded = batchify.Pad(pad_val=-1)([mx.nd.array([]), mx.nd.arange(1)]).asnumpy().flatten().tolist()
     assert padded == [-1.0, 0.0]
 
+
 def test_stack_batchify():
-    batchify_fn = batchify.Stack()
     for dtype in [np.uint8, np.int32, np.int64, np.float16, np.float32, np.float64]:
         dat = [np.random.randint(5, size=(10,)).astype(dtype) for _ in range(10)]
-        batchify_out = batchify_fn(dat).asnumpy()
-        npy_out = np.array(dat)
-        assert_allclose(batchify_out, npy_out)
-        assert batchify_out.dtype == npy_out.dtype
+        for _dtype in [None, dtype]:
+            batchify_fn = batchify.Stack(dtype=_dtype)
+            batchify_out = batchify_fn(dat).asnumpy()
+            npy_out = np.array(dat)
+            assert_allclose(batchify_out, npy_out)
+            assert batchify_out.dtype == npy_out.dtype
```

Comment:
It has not checked the case that the `dtype` used in Stack is different from the `dtype` of the input array. We may do the following instead:
```python
for _dtype in [np.float32]:
     batchify_fn = batchify.Stack(dtype=_dtype)
     batchify_out = batchify_fn(dat).asnumpy()
     assert batchify_out.dtype == np.float32
```